### PR TITLE
Fix file enconding

### PR DESCRIPTION
**Pull request by rainbow-six3**
With #74 i fixed the writing of the snapshot file with correct encoding. Now ther is a sencond problem when reading the existing snapshotfile, that i fixed. Yaaay

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,10 +12,16 @@ subprojects { subproject ->
...
+    subproject.apply plugin: 'java'
 
     sourceCompatibility = '1.8'
     targetCompatibility = '1.8'
 
+    compileJava.options.encoding = 'UTF-8'
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+    }
+
```
Comment (jackmatt2) [inline build.gradle:24]: Why is this needed?  Ran tests on my machine without it and they all passed.

File: build.gradle [hunk 2]
```diff
@@ -12,10 +12,16 @@ subprojects { subproject ->
     subproject.apply plugin: 'java-library'
     subproject.apply plugin: 'com.github.johnrengelman.shadow'
     subproject.apply plugin: 'maven-publish'
+    subproject.apply plugin: 'java'
```
Comment (jackmatt2) [inline build.gradle:15]: This can go too if the UTF-8 gradle config is not needed.

File: java-snapshot-testing-core/src/test/java/au/com/origin/snapshots/onLoadSnapshotFileTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+public class onLoadSnapshotFileTest {
```
Comment (jackmatt2) [inline java-snapshot-testing-core/src/test/java/au/com/origin/snapshots/onLoadSnapshotFileTest.java:20]: Can you rename to `OnLoadSnapshotFileTest`, I'm not sure why `onSaveSnapshotFileTest.java` is not following conventions.  It needs to be cleaned up.

File: build.gradle (hunk 1 above)
Comment (rainbow-six3) [reply build.gradle:24]: Okay. Sound fine for me. I had just applied  the settings like in fromhttps://stackoverflow.com/questions/38390206/assertj-wrong-failure-when-comparing-two-unicode-strings

File: java-snapshot-testing-core/src/test/java/au/com/origin/snapshots/onLoadSnapshotFileTest.java (hunk 3 above)
Comment (rainbow-six3) [reply java-snapshot-testing-core/src/test/java/au/com/origin/snapshots/onLoadSnapshotFileTest.java]: Done

File: build.gradle (hunk 1 above)
Comment (jackmatt2) [reply build.gradle:24]: Yeah, I think this will mask the issue we are resolving here.  That is - the code should work regardless of the encoding of the underlying OS.

Comment (rainbow-six3) [reply build.gradle:24]: When i remove the config the test fails. (i work with a windows machine)

Comment (jackmatt2) [reply build.gradle:24]: Particular tests?  I'm on a mac.

Comment (jackmatt2) [reply build.gradle:24]: If it's a test - perhaps the writing of the "test" files needs to use UTF-8.

Comment (jackmatt2) [reply build.gradle:24]: I run agains a Windows machine on CI too.  Would be interesting to see if it also fails there.

Comment (rainbow-six3) [reply build.gradle:20]: I was my bad. my test was wrong. i removed the config

File: build.gradle (hunk 2 above)
Comment (rainbow-six3) [reply build.gradle]: Removed.

File: build.gradle [hunk 4]
```diff
@@ -16,6 +16,8 @@ subprojects { subproject ->
     sourceCompatibility = '1.8'
     targetCompatibility = '1.8'
 
+    compileJava.options.encoding = 'UTF-8'
```
Comment (jackmatt2) [inline build.gradle:19]: this still needs to go

Comment (rainbow-six3) [reply build.gradle]: Done. I hope now i have every thing alright.

**Review by jackmatt2 — APPROVED** [review]
Looks good thanks - will deploy 3.2.2 shortly
